repository/maven.go

```diff
@@ -155,35 +148,66 @@ func (r *Maven) writeSettings() (path string, err error) {
 			path)
 		return
 	}
-	settings := id.Settings
-	settings, err = r.injectProxy(id)
+	defer f.Close()
+	id, found, err := r.findIdentity("maven")
 	if err != nil {
 		return
 	}
-	_, err = f.Write([]byte(settings))
+	settingsXML, err := mxj.NewMapXml([]byte(emptySettings))
+	if err != nil {
+		err = liberr.Wrap(err)
+		return "", err
+	}
+	if found {
+		addon.Activity(
+			"[MVN] Using credentials (id=%d) %s.",
+			id.ID,
+			id.Name)
+		// THis will use the settings in the idendity for settingsXML
+		settingsXML, err = r.injectProxy(id)
+		if err != nil {
+			return
+		}
+	}
+	settingsXML, err = r.injectCacheDir(settingsXML)
+	if err != nil {
+		return
+	}
+	settings, err := settingsXML.XmlIndent("", "  ")
+	if err != nil {
+		err = liberr.Wrap(err)
+		return "", err
+	}
+	_, err = f.Write(settings)
 	if err != nil {
 		err = liberr.Wrap(
 			err,
 			"path",
 			path)
 	}
-	_ = f.Close()
-	addon.Activity("[FILE] Created %s.", path)
+	addon.Activity("[FILE] Created %s. -- file: %s", path, settings)
```

Comment:
We cannot include the settings file in the task activity log because it contains credentials.